### PR TITLE
Remove cogs position from redux

### PR DESCRIPTION
**Pull request by msmanoj**
## Description
- Removed the cogs position from redux and moved it to `useBrowserCogList`
-  Moved `selectedCog` one level up in redux
- Updated the tests

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1616

## Deployment URL(s)
http://cogs-out-of-redux.review.ensembl.org

## Views affected
Cogs

### CODE REVIEW

File: src/content/app/genome-browser/components/browser-cog/useBrowserCogList.ts [hunk 1]
```diff
@@ -33,7 +33,8 @@ import {
   getTrackType,
   TrackType,
   type TrackSettings,
-  type TrackSettingsForGenome
+  type TrackSettingsForGenome,
+  CogList
```
Comment (azangru) [inline src/content/app/genome-browser/components/browser-cog/useBrowserCogList.ts:37]: `type`?

Comment (msmanoj) [reply src/content/app/genome-browser/components/browser-cog/useBrowserCogList.ts]: That's done

**Comment by azangru** [comment]
Could you merge the latest dev, and check whether it's something in your PR that broke 1/5 transcripts, or it was something in the old dev.

https://user-images.githubusercontent.com/6834224/182581683-5f40c58c-13d6-4b22-86a4-daf828b44a10.mov

**Comment by msmanoj** [comment]
> Could you merge the latest dev, and check whether it's something in your PR that broke 1/5 transcripts, or it was something in the old dev.
> 
It's something I broke in this PR. I'll take a look

**Comment by msmanoj** [comment]
> > Could you merge the latest dev, and check whether it's something in your PR that broke 1/5 transcripts, or it was something in the old dev.
> 
> It's something I broke in this PR. I'll take a look

This is now fixed. The problem was that the state of the `cogList` was not shared between trackSettingsPanel and cogs.

**Comment by azangru** [comment]
> The problem was that the state of the `cogList` was not shared between trackSettingsPanel and cogs.

image

Confused: why is it needed on the browser context?

**Comment by msmanoj** [comment]
> > The problem was that the state of the `cogList` was not shared between trackSettingsPanel and cogs.
> 
> image
> 
> Confused: why is it needed on the browser context?

It needs to be in some context so that the state of the cogList is shared between different places where the `useBrowserCogList` is used. I thought it would be better to add to the existing browser context rather than creating another one.

**Comment by msmanoj** [comment]
Another solution will be is to make the `useBrowserCogList` hook directly listen to the cog updates payload which I am not sure if it'll be better.

**Comment by azangru** [comment]
> It needs to be in some context so that the state of the cogList is shared between different places where the `useBrowserCogList` is used. I thought it would be better to add to the existing browser context rather than creating another one.

I'm trying to decide whether passing the cog list through the genome browser context is significantly better than just passing it through props  (BrowserCogList -> BrowserCog -> TrackPanelSettings). Normally, I would go for props. Is there a stronger argument in favour of the context?

**Comment by msmanoj** [comment]
> I'm trying to decide whether passing the cog list through the genome browser context is significantly better than just passing it through props (BrowserCogList -> BrowserCog -> TrackPanelSettings). Normally, I would go for props. Is there a stronger argument in favour of the context?

Not an issue now but passing it as props will restrict the hierarchy of these three components which will be an issue if we decide to separate out `TrackPanelSettings` from `BrowserCog`. Also, we'll need something like a context if we need to access `cogList` outside of `BrowserCogList`. Again, it's not an issue now. 
I think using context is fine here as it doesn't create any clutter but I don't have any preference.

**Comment by msmanoj** [comment]
> Not an issue now but passing it as props will restrict the hierarchy of these three components which will be an issue if we decide to separate out `TrackPanelSettings` from `BrowserCog`. Also, we'll need something like a context if we need to access `cogList` outside of `BrowserCogList`. Again, it's not an issue now. I think using context is fine here as it doesn't create any clutter but I don't have any preference.

I also noticed that it won't be easy for `useBrowserTrackSettings` to access the cogList if we decide to pass it as props.

**Comment by azangru** [comment]
> I also noticed that it won't be easy for `useBrowserTrackSettings` to access the cogList if we decide to pass it as props.

If we did that, we would pass the cogs list as another argument to the functions called from TrackPanelSettings

**Comment by msmanoj** [comment]
> > I also noticed that it won't be easy for `useBrowserTrackSettings` to access the cogList if we decide to pass it as props.
> 
> If we did that, we would pass the cogs list as another argument to the functions called from TrackPanelSettings

This would require passing the props further down to `RegularTrackSettings` & `GeneTrackSettings`.

**Review by azangru — APPROVED** [review]

**Pull request merged**